### PR TITLE
Gh 33 context class

### PR DESCRIPTION
**Pull request by GCHQDeveloper404**

### CODE REVIEW

File: common/src/main/java/uk/gov/gchq/palisade/Context.java [hunk 1]
```diff
@@ -16,22 +16,40 @@
...
 
-public class Justification {
-    private String justification;
+import java.util.HashMap;
+import java.util.Map;
 
-    public Justification() {
+@JsonPropertyOrder(value = {"class", "context"}, alphabetic = true)
+public class Context {
+    private Map<String, Object> context;
```
Comment (ac74475) [inline common/src/main/java/uk/gov/gchq/palisade/Context.java]: Can we use a different name to prevent confusion with the class name?

Comment (ac74475) [reply common/src/main/java/uk/gov/gchq/palisade/Context.java]: also the context could just extend HashMap<String, Object>, that way you already have direct access to the getValue, getKeySet methods without first getting the HashMap out of the context object

File: common/src/main/java/uk/gov/gchq/palisade/policy/Rule.java [hunk 2]
```diff
@@ -55,10 +55,10 @@
      *
      * @param record        the record to be checked.
      * @param user          the user
-     * @param justification the query justificaiton
+     * @param context the query justificaiton
```
Comment (ac74475) [inline common/src/main/java/uk/gov/gchq/palisade/policy/Rule.java]: update the comment

File: common/src/main/java/uk/gov/gchq/palisade/policy/Rule.java [hunk 3]
```diff
@@ -22,14 +22,14 @@
...
 
 /**
  * <p>
  * A {@code Rule} is the fundamental interface for applying policy criteria.
  * It allows a record to redacted or modified based on a {@link User} and their
- * query {@link Justification}.
+ * query {@link Context}.
  * </p>
  * <p>
  * Please note, the justification is optional and may be null.
```
Comment (ac74475) [inline common/src/main/java/uk/gov/gchq/palisade/policy/Rule.java]: update the comment

File: service/policy-service/src/main/java/uk/gov/gchq/palisade/policy/service/request/CanAccessRequest.java [hunk 4]
```diff
@@ -159,7 +159,7 @@ public String toString() {
                 .appendSuper(super.toString())
                 .append("user", user)
                 .append("resource", resource)
-                .append("justification", justification)
+                .append("justification", context)
```
Comment (ac74475) [inline service/policy-service/src/main/java/uk/gov/gchq/palisade/policy/service/request/CanAccessRequest.java]: change the string name

File: service/policy-service/src/main/java/uk/gov/gchq/palisade/policy/service/request/GetPolicyRequest.java [hunk 5]
```diff
@@ -127,7 +127,7 @@ public String toString() {
         return new ToStringBuilder(this)
                 .appendSuper(super.toString())
                 .append("user", user)
-                .append("justification", justification)
+                .append("justification", context)
```
Comment (ac74475) [inline service/policy-service/src/main/java/uk/gov/gchq/palisade/policy/service/request/GetPolicyRequest.java]: change the name in the string

File: common/src/main/java/uk/gov/gchq/palisade/Context.java (hunk 1 above)
Comment (GCHQDeveloper404) [reply common/src/main/java/uk/gov/gchq/palisade/Context.java]: context -> contents

**Pull request closed without merging**